pkg/controllers/machinesetsync/machineset_sync_controller.go

```diff
@@ -384,15 +399,28 @@ func (r *MachineSetSyncReconciler) reconcileCAPIMachineSetToMAPIMachineSet(ctx c
 		r.Recorder.Event(mapiMachineSet, corev1.EventTypeWarning, "ConversionWarning", warning)
 	}
 
-	newMapiMachineSet.Spec.Template.Labels = util.MergeMaps(mapiMachineSet.Spec.Template.Labels, newMapiMachineSet.Spec.Template.Labels)
+	newMapiMachineSet.SetGeneration(mapiMachineSet.GetGeneration())
+	newMapiMachineSet.SetUID(mapiMachineSet.GetUID())
+	newMapiMachineSet.SetCreationTimestamp(mapiMachineSet.GetCreationTimestamp())
+	newMapiMachineSet.SetManagedFields(mapiMachineSet.GetManagedFields())
+	newMapiMachineSet.SetResourceVersion(util.GetResourceVersion(client.Object(mapiMachineSet)))
 	newMapiMachineSet.SetNamespace(mapiMachineSet.GetNamespace())
-	// The conversion does not set a resource version, so we must copy it over
-	newMapiMachineSet.SetResourceVersion(util.GetResourceVersion(mapiMachineSet))
-	newMapiMachineSet.OwnerReferences = []metav1.OwnerReference{} // No CAPI machine set owner references are converted to MAPI machine set.
+	newMapiMachineSet.Spec.Template.ObjectMeta.Labels = util.MergeMaps(mapiMachineSet.Spec.Template.ObjectMeta.Labels, newMapiMachineSet.Spec.Template.ObjectMeta.Labels)
```

Comment:
I can't remember, did we decide that we want CAPI labels to be propagated to the MAPI counterpart?

Reply:
But now we might even be able to drop this. I'll check.